tests/rados/pool_tests.py

```diff
@@ -995,73 +996,112 @@ def run(ceph_cluster, **kw):
+                    raise Exception("Reads failed on the cluster Error")
+
+                if pool_obj.do_rados_put(client=client_node, pool=pool_name, nobj=10):
+                    log.error(
+                        "UnAble to perform writes at min_size with 1 OSD down. Error"
+                    )
+                    raise Exception("Writes failed on the cluster Error")
+
+                log.info(
+                    "Able to perform read & writes operations on the cluster"
+                    " Proceeding to bring down another OSD on the PG"
+                )
+
                 if not rados_obj.change_osd_state(
-                    action="stop", target=second_osd_pick, timeout=20
+                    action="stop", target=second_osd_pick, timeout=200
                 ):
                     log.error(f"Failed to stop OSD : {second_osd_pick}")
                     raise Exception("test-bed set-up failure")
                 time.sleep(10)
 
                 # Get the min_size property of the pool
-                min_size_val = 0
                 min_size_val = rados_obj.get_pool_property(
                     pool=pool_name, props="min_size"
                 )
                 log.info(f"Current min_size of the pool - {min_size_val}")
                 post_pg_set = rados_obj.get_pg_acting_set(pg_num=pgid)
                 log.info(
-                    f"Acting set post OSDs are down and before rados put op : {post_pg_set}"
+                    f"Acting set post 2 OSDs are down and before rados put op : {post_pg_set}"
                 )
-                log.info(f"Current OSDs in acting set {post_pg_set}")
+                if len(post_pg_set) != 1:
+                    log.error("Acting set contains more than 1 OSD. Fail")
+                    raise Exception("Acting set changed error")
+
+                pool_stat_1 = rados_obj.get_cephdf_stats(pool_name=pool_name)
+                total_objs_1 = pool_stat_1["stats"]["objects"]
+                log.debug(
+                    f"pool stats when 2 OSDs from AC is down before write is {pool_stat_1}. "
+                    f"\n Total objects on the pool {total_objs_1}"
+                )
+
+                if pool_obj.do_rados_get(pool=pool_name, read_count=20):
+                    log.error(f"Could perform read operations on the pool {pool_name}")
+                    raise Exception("Reads failed on the cluster Error")
+
+                if not pool_obj.do_rados_put(
+                    client=client_node, pool=pool_name, nobj=5
+                ):
+                    log.error("Able to perform writes at min_size. Error")
+                    raise Exception("Writes failed on the cluster Error")
+
                 log.info(
-                    f"Initial pg set fetched from pool id {pool_id} for pg id {pgid} is {pg_set}"
+                    "Unable to perform read & write operations on the cluster as expected"
+                    " Proceeding to bring down another OSD on the PG"
```

Comment:
At this point the PG has only 1 OSD, are we going to bring down that OSD as well?

Reply:
No. at this point, only 1 OSD is down.